functions/mqhandlers.go

```diff
@@ -280,6 +280,8 @@ func handleEndpointDetection(peerUpdate *models.HostPeerUpdate) {
 				peerIface := peerInfo.Interfaces[i]
 				peerIP := peerIface.Address.IP
 				if strings.Contains(peerIP.String(), "127.0.0.") ||
+					peerIP.IsMulticast() ||
+					(peerIP.IsLinkLocalUnicast() && strings.Count(peerIP.String(), ":") >= 2) ||
```

Comment:
this just masks out ipv6 addresses so ipv6 endpoints are not checked.  To enable ipv6 checking, the first param to FindBest endpoint neeeds to be enclosed in "[ ]" if it ipv6 or FindBestEndpoint should update the reqAddr if it ipv6 before calling net.DialTimeout

Reply:
> this just masks out ipv6 addresses so ipv6 endpoints are not checked.  To enable ipv6 checking, the first param to FindBest endpoint neeeds to be enclosed in "[ ]" if it ipv6 or FindBestEndpoint should update the reqAddr if it ipv6 before calling net.DialTimeout

Ipv6 addresses are still checked. The [] enclosure is added in the find best endpoint function if you look at the changes there. Even after adding the enclosure, errors were occurring because ipv6 link-local unicast addresses additionally require specifying the interface to use when dialing, which is why I'm skipping that specific type of addres here.